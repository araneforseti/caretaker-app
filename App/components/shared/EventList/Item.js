import React from 'react';
import {
  View,
  TouchableOpacity
} from 'react-native';
import DateDisplay from '../DateDisplay';
import EventSummaryBox from '../EventSummaryBox';

import styles from './Item.styles';

export default class Item extends React.PureComponent {
  _onPress = () => this.props.onPressItem(this.props.id);

  render() {
    const {
      startDate,
      name,
      volunteer
    } = this.props;

    return <TouchableOpacity onPress={this._onPress}>
      <View style={styles.container}>
        <DateDisplay date={startDate} />
        <EventSummaryBox name={name} volunteer={volunteer} />
      </View>
    </TouchableOpacity>;
  }
}
