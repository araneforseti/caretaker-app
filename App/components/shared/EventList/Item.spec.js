import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

describe('<EventList.Item />', () => {
  const Jul_10_2017 = new Date(1499662800000);
  const Jul_11_2017 = new Date(1499749200000);

  let props;
  const item = () => renderer.create(<Item {...props} />);

  beforeEach(() => {
    props = {
      id: 1,
      onPressItem: jest.fn(),
      title: 'Title',
      startDate: Jul_10_2017,
      endDate: Jul_11_2017,
      date: 'Jul 10',
      name: 'Chemotherapy',
      volunteer: 'The Dude'
    };
  });

  it('should render correctly', () => {
    expect(item()).toMatchSnapshot();
  });

});
