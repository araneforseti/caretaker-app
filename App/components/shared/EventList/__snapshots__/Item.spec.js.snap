// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventList.Item /> should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
          "width": 56,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        10
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}
      >
        Mon
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CC5D5D",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "padding": 4,
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#CC5D5D",
            },
            Object {
              "fontWeight": "bold",
            },
            false,
          ]
        }
      >
        Chemotherapy
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#CC5D5D",
            },
            Object {},
            false,
            false,
          ]
        }
      >
        The Dude
      </Text>
    </View>
  </View>
</View>
`;
